packages/app-layout/AppLayout.tsx

```diff
@@ -0,0 +1,161 @@
+import { h, Component, prop } from 'skatejs';
+
+import {
+  GenericTypes,
+  matchMedia
+} from '../_helpers';
+
+import { HamburgerButton } from './components/HamburgerButton';
+import { Drawer } from './components/Drawer';
+import { Overlay } from './components/Overlay';
+import { Nav } from './components/Nav';
+import { NavContent } from './components/NavContent';
+
+import styles from './AppLayout.scss';
+
+const DEFAULT_RESPONSIBLE_WIDTH = '640px';
+
+type AppLayoutProps = Props & EventProps;
+
+type Attrs = {
+  'drawer-visible'?: boolean,
+  'responsive-width'?: string
+};
+
+type Props = {
+  drawerVisible?: boolean,
+  responsiveWidth?: string
+};
+
+// TODO: add onDrawerChange / onNarrowChange
+type EventProps = {};
+
+type Events = {};
+
+declare global {
+  namespace JSX {
+    interface IntrinsicElements {
+      'bl-app-layout': GenericTypes.IntrinsicCustomElement<AppLayoutProps>
+      & GenericTypes.IntrinsicBoreElement<Attrs, Events>
+    }
+  }
+}
+
+export class AppLayout extends Component<AppLayoutProps> {
+
+  static get is() { return 'bl-app-layout'; }
+
+  static get props() {
+    return {
+      drawerVisible: prop.boolean( {
+        attribute: {
+          source: true
+        }
+      } ),
+      responsiveWidth: prop.string( {
+        initial: DEFAULT_RESPONSIBLE_WIDTH,
+        attribute: {
+          source: true
+        }
+      } ),
+      narrow: prop.boolean( {
+        attribute: {
+          source: true,
```

Comment:
> I haven't read your code, but thing to consider ( I've already discussed it with Dave. ) Do we really need this to be declared via prop API ( it makes only sense if you wanna really ? trigger re render. ) If don't rather use it directly on instance and if you are mutating props that are defined via static props get API, always use props ( which is setState equivalent in React ).

yep that's the case, I need to rerender when this property change.

> Also when setting via props function updates are batched so they are predictable. Instead of you set directly to this, it is asynchronous operation. In skate we are going to strictly enforce using props in next release. Don't hasitate to ask further. Cheers!

ups did miss this pattern, will fix that. If I may I would suggest to rather change naming to either setProps or even setState as props feels too ambiguous.